logstash-core/lib/logstash/api/lib/app/commands/system/basicinfo_command.rb

```diff
@@ -0,0 +1,31 @@
+# encoding: utf-8
+require "app/command"
+require "logstash/util/duration_formatter"
+
+class LogStash::Api::SystemBasicInfoCommand < LogStash::Api::Command
+
+  attr_reader :agent
+
+  def run
+    @agent = service.agent
+    {
+      "version"   => LOGSTASH_VERSION,
+      "hostname" => @agent.node_name,
+    }
+  end
```

Comment:
There is some conflicting codes with the reference to `agent`, first we declare an `attr_reader :agent`, the value is set in the `#run` method but we never use it. We always use the *instance variable*.

Could we make the `agent`  handling part of the `LogStash::Api::Command` base class?
Could this be useful to other command?